src/app/api/file/route.schema.ts

```diff
@@ -0,0 +1,66 @@
+import { z } from "zod";
+import { unauthorizedErrorSchema, unknownErrorSchema } from "@api/route.schema";
+/*
+  id           String   @id @default(auto()) @map("_id") @db.ObjectId
+  date         DateTime // will restrict hours to midnight 
+  filetype     String
+  url          String
+  seniorId     String   @db.ObjectId
+  senior       Senior   @relation(fields: [seniorId], references: [id], onDelete: Cascade)
+  Tags         String[]
+
+  @@unique([seniorId, date])
+*/
+
+export const File = z.object({
+  date: z.string().transform((val) => {
+    const date = new Date(val);
+    date.setHours(0, 0, 0, 0);
+    return date;
+  }),
+  filetype: z.string(),
+  url: z.string(),
+  Tags: z.array(z.string()),
+  seniorId: z.string(),
+});
+
+export const FileResponse = z.discriminatedUnion("code", [
+  z.object({
+    code: z.literal("SUCCESS"),
+    message: z.literal("File successfully added"),
+  }),
+  z.object({
+    code: z.literal("INVALID_FILE"),
+    message: z.literal("Invalid file added"),
+  }),
+  z.object({
+    code: z.literal("UNKNOWN"),
+    message: z.literal("Unknown error received"),
+  }),
+  z.object({
+    code: z.literal("NOT_AUTHORIZED"),
+    message: z.literal("Senior not assigned to user"),
+  }),
+  z.object({
+    code: z.literal("NO_SENIOR"),
+    message: z.literal("Senior does not exist"),
+  }),
+  z.object({
+    code: z.literal("NO_FILE"),
+    message: z.literal("File does not exist"),
+  }),
+  z.object({
+    code: z.literal("NO_USER"),
```

Comment:
I think we can assume that user exists if they're authorized, so we can eliminate this case.